server/store/store_test.go

```diff
@@ -150,3 +189,228 @@ func TestCheckEnabledTeamByTeamID(t *testing.T) {
 		})
 	}
 }
+
+func TestStoreChannelLinkAndGetLinkByChannelID(t *testing.T) {
+	assert := assert.New(t)
+	store, api, container := setupTestStore(&plugintest.API{})
+	store.enabledTeams = func() []string { return []string{"mockMattermostTeamID"} }
+
+	api.On("GetTeam", "mockMattermostTeamID").Return(&model.Team{
+		Name: "mockMattermostTeamID",
+	}, nil)
+
+	mockChannelLink := &storemodels.ChannelLink{
+		MattermostChannel: "mockMattermostChannelID",
+		MattermostTeam:    "mockMattermostTeamID",
+		MSTeamsTeam:       "mockMSTeamsTeamID",
+		MSTeamsChannel:    "mockMSTeamsChannelID",
+		Creator:           "mockCreator",
+	}
+
+	storeErr := store.StoreChannelLink(mockChannelLink)
+	assert.Nil(storeErr)
+
+	resp, getErr := store.GetLinkByChannelID("mockMattermostChannelID")
+	assert.Equal(mockChannelLink, resp)
+	assert.Nil(getErr)
+
+	termErr := container.Terminate(context.Background())
```

Comment:
Of course if you follow the tear down approach you no longer need this two lines here